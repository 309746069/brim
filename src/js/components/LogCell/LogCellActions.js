/* @flow */

import React from "react"
import Field from "../../models/Field"
import Log from "../../models/Log"
import {ContextMenu, MenuItem} from "../ContextMenu"
import buildMenu from "../LogCell/buildMenu"
import type {Space} from "../../lib/Space"
import type {MenuItemData} from "../LogCell/buildMenu"

export type Props = {
  field: Field,
  log: Log,
  style: Object,
  space: Space,
  onClose: Function,
  dispatch: Function
}

export default class LogCellActions extends React.Component<Props> {
  menu: MenuItemData[]

  constructor(props: Props) {
    super(props)
    this.menu = buildMenu(props)
  }

  renderItem(item: MenuItemData, index: number) {
    switch (item.type) {
      case "action":
        return (
          <MenuItem key={index} onClick={item.onClick}>
            {item.text}
          </MenuItem>
        )
      case "seperator":
        return <hr key={index} />
    }
  }

  render() {
    return (
      <ContextMenu onOutsideClick={this.props.onClose} style={this.props.style}>
        {this.menu.map(this.renderItem)}
      </ContextMenu>
    )
  }
}
