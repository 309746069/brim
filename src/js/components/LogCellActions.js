/* @flow */

import React from "react"
import Field from "../models/Field"
import Log from "../models/Log"
import {ContextMenu, MenuItem} from "./ContextMenu"
import * as searchBar from "../actions/searchBar"
import * as packets from "../actions/packets"
import type {Space} from "../lib/Space"

type Props = {
  field: Field,
  log: Log,
  style: Object,
  space: Space,
  onClose: Function,
  dispatch: Function
}

type Action = {
  type: "action",
  text: string,
  onClick: Event => void
}

type Seperator = {
  type: "seperator"
}

type MenuItemData = Seperator | Action

export default class LogCellActions extends React.Component<Props> {
  menu: MenuItemData[]

  constructor(props: Props) {
    super(props)
    const {dispatch, field, log, space} = props

    this.menu = [
      {
        type: "action",
        text: "Filter out these values",
        onClick: (e: Event) => {
          e.stopPropagation()
          dispatch(searchBar.appendQueryExclude(field))
          dispatch(searchBar.submitSearchBar())
        }
      },
      {
        type: "action",
        text: "Only show these values",
        onClick: (e: Event) => {
          e.stopPropagation()
          dispatch(searchBar.appendQueryInclude(field))
          dispatch(searchBar.submitSearchBar())
        }
      },
      {
        type: "action",
        text: `Count by ${field.name}`,
        onClick: (e: Event) => {
          e.stopPropagation()
          dispatch(searchBar.appendQueryCountBy(field))
          dispatch(searchBar.submitSearchBar())
        }
      }
    ]

    if (log.isPath("conn") && space.packet_support) {
      this.menu = this.menu.concat([
        {
          type: "seperator"
        },
        {
          type: "action",
          text: "Download PCAPS",
          onClick: (_e: Event) => {
            dispatch(packets.fetchPackets(log))
          }
        }
      ])
    }
  }

  renderItem(item: MenuItemData, index: number) {
    switch (item.type) {
      case "action":
        return (
          <MenuItem key={index} onClick={item.onClick}>
            {item.text}
          </MenuItem>
        )
      case "seperator":
        return <hr key={index} />
    }
  }

  render() {
    return (
      <ContextMenu onOutsideClick={this.props.onClose} style={this.props.style}>
        {this.menu.map(this.renderItem)}
      </ContextMenu>
    )
  }
}
