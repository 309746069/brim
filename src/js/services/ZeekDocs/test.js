/* @flow */
import ZeekDocs from "./"

test("describe conn uid", () => {
  let desc = ZeekDocs.describe("conn", "uid")

  expect(desc).toBe("A unique identifier of the connection.")
})
