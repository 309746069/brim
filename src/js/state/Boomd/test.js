/* @flow */
import Boomd from "./"
import initTestStore from "../../test/initTestStore"

test("setting use analytics cache", () => {
  const store = initTestStore()
  store.dispatch(Boomd.useCache(true))
  expect(Boomd.usingCache(store.getState())).toEqual(true)
  store.dispatch(Boomd.useCache(false))
  expect(Boomd.usingCache(store.getState())).toEqual(false)
})

test("setting use index", () => {
  const store = initTestStore()
  store.dispatch(Boomd.useIndex(true))
  expect(Boomd.usingIndex(store.getState())).toEqual(true)
  store.dispatch(Boomd.useIndex(false))
  expect(Boomd.usingIndex(store.getState())).toEqual(false)
})
