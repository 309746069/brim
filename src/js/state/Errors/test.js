/* @flow */

import AppError from "../../models/AppError"
import Errors from "./"
import initTestStore from "../../test/initTestStore"

let store
beforeEach(() => {
  store = initTestStore()
})

test("ERROR_CREATE", () => {
  let state = store.dispatchAll([Errors.createError("Bug")])

  expect(Errors.getErrors(state)[0]).toEqual(expect.any(AppError))
})

test("ERRORS_CLEAR", () => {
  let state = store.dispatchAll([
    Errors.createError("Bug"),
    Errors.createError("Bug2"),
    Errors.clearErrors()
  ])

  expect(Errors.getErrors(state)).toEqual([])
})
