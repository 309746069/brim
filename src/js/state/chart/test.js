/* @flow */

import chart from "./"
import initTestStore from "../../test/initTestStore"

let store
beforeEach(() => {
  store = initTestStore()
})

let records = [
  [{name: "id", type: "string", value: "1"}],
  [{name: "id", type: "string", value: "2"}]
]

test("chart records append", () => {
  let state = store.dispatchAll([
    chart.appendRecords([records[0]]),
    chart.appendRecords([records[1]])
  ])

  expect(chart.getRecords(state)).toEqual(records)
})

test("chart records status", () => {
  let state = store.dispatchAll([chart.setStatus("SUCCESS")])

  expect(chart.getStatus(state)).toBe("SUCCESS")
})

test("chart records clear", () => {
  let state = store.dispatchAll([
    chart.appendRecords(records),
    chart.setStatus("SUCCESS"),
    chart.clear()
  ])

  expect(chart.getStatus(state)).toBe("INIT")
  expect(chart.getRecords(state)).toEqual([])
})
