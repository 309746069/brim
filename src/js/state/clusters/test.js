/* @flow */

import {addCluster, removeCluster, setCluster, setClusterError} from "./actions"
import {
  getCurrentCluster,
  getClusterMessage,
  getSavedClusters
} from "./selectors"
import initTestStore from "../../test/initTestStore"

let store
beforeEach(() => {
  store = initTestStore()
})

let cluster = {
  host: "boom.com",
  port: "9867",
  username: "kerr",
  password: "123"
}

test("addCluster", () => {
  let state = store.dispatchAll([addCluster(cluster)])

  expect(getSavedClusters(state)).toEqual([cluster])
})

test("addCluster when it already exists", () => {
  let state = store.dispatchAll([addCluster(cluster), addCluster(cluster)])

  expect(getSavedClusters(state)).toEqual([cluster])
})

test("removeCluster", () => {
  let state = store.dispatchAll([addCluster(cluster), removeCluster(cluster)])

  expect(getSavedClusters(state)).toEqual([])
})

test("setCluster", () => {
  let state = store.dispatchAll([setCluster(cluster)])

  expect(getCurrentCluster(state)).toEqual(cluster)
})

test("setClusterMessage to message", () => {
  let state = store.dispatchAll([setClusterError("no spaces")])

  expect(getClusterMessage(state)).toEqual("no spaces")
})

test("setClusterMessage to empty", () => {
  let state = store.dispatchAll([setClusterError("")])

  expect(getClusterMessage(state)).toEqual("")
})
