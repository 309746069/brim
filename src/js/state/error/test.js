/* @flow */
import AppError from "../../models/AppError"
import error from "./"
import initTestStore from "../../test/initTestStore"

let store
beforeEach(() => {
  store = initTestStore()
})

test("init state", () => {
  let state = store.getState()
  expect(error.getError(state)).toEqual(null)
  expect(error.getVisible(state)).toBe(false)
})

test("set", () => {
  let e = new AppError()
  let state = store.dispatchAll([error.set(e)])

  expect(error.getError(state)).toEqual(e)
  expect(error.getVisible(state)).toBe(true)
})

test("clear", () => {
  let e = new AppError()
  let state = store.dispatchAll([error.set(e), error.clear()])

  expect(error.getError(state)).toEqual(null)
  expect(error.getVisible(state)).toBe(true)
})

test("dismiss", () => {
  let e = new AppError()
  let state = store.dispatchAll([error.set(e), error.dismiss()])

  expect(error.getError(state)).toEqual(e)
  expect(error.getVisible(state)).toBe(false)
})
