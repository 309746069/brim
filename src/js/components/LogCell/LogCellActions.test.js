/* @flow */

import React from "react"
import {shallow} from "enzyme"
import LogCellActions from "./LogCellActions"
import * as mockLogs from "../../test/mockLogs"
import mockSpace from "../../test/mockSpace"

test("basic snapshot", () => {
  const log = mockLogs.conn()
  const space = mockSpace({packet_support: true})
  const wrapper = shallow(
    <LogCellActions
      menuActions={[]}
      resultType="logs"
      dispatch={jest.fn()}
      log={log}
      field={log.getField("_path")}
      space={space}
      onClose={() => {}}
      style={{top: 1, left: 1}}
    />
  )
  expect(wrapper).toMatchSnapshot()
})
