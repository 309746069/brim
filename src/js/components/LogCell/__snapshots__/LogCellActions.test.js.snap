// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LogCellActions
    dispatch={[MockFunction]}
    field={
      Field {
        "name": "_path",
        "type": "string",
        "value": "conn",
      }
    }
    log={
      Log {
        "descriptor": Array [
          Object {
            "name": "_td",
            "type": "int",
          },
          Object {
            "name": "_path",
            "type": "string",
          },
          Object {
            "name": "ts",
            "type": "time",
          },
          Object {
            "name": "uid",
            "type": "string",
          },
          Object {
            "name": "id.orig_h",
            "type": "addr",
          },
          Object {
            "name": "id.orig_p",
            "type": "port",
          },
          Object {
            "name": "id.resp_h",
            "type": "addr",
          },
          Object {
            "name": "id.resp_p",
            "type": "port",
          },
          Object {
            "name": "proto",
            "type": "enum",
          },
          Object {
            "name": "service",
            "type": "string",
          },
          Object {
            "name": "duration",
            "type": "interval",
          },
          Object {
            "name": "orig_bytes",
            "type": "count",
          },
          Object {
            "name": "resp_bytes",
            "type": "count",
          },
          Object {
            "name": "conn_state",
            "type": "string",
          },
          Object {
            "name": "local_orig",
            "type": "bool",
          },
          Object {
            "name": "local_resp",
            "type": "bool",
          },
          Object {
            "name": "missed_bytes",
            "type": "count",
          },
          Object {
            "name": "history",
            "type": "string",
          },
          Object {
            "name": "orig_pkts",
            "type": "count",
          },
          Object {
            "name": "orig_ip_bytes",
            "type": "count",
          },
          Object {
            "name": "resp_pkts",
            "type": "count",
          },
          Object {
            "name": "resp_ip_bytes",
            "type": "count",
          },
          Object {
            "name": "tunnel_parents",
            "type": "set[string]",
          },
        ],
        "tuple": Array [
          "0",
          "conn",
          "1425612054.369843",
          "Cynwae4qh1GxM82hQ2",
          "192.168.0.50",
          "1900",
          "239.255.255.250",
          "1900",
          "udp",
          "-",
          "2.000293",
          "282",
          "0",
          "S0",
          "-",
          "-",
          "0",
          "D",
          "3",
          "366",
          "0",
          "0",
          "(empty)",
        ],
      }
    }
    onClose={[Function]}
    space={
      Object {
        "close_timeout": 5000,
        "compression": "none",
        "flush_timeout": 500,
        "maxTime": "2015-04-13 09:36:33.750",
        "max_time": Object {
          "ns": 750000000,
          "sec": 1428917793,
        },
        "max_writers": 150,
        "minTime": "2015-03-05 14:15:00.000",
        "min_time": Object {
          "ns": 0,
          "sec": 1425564900,
        },
        "name": "default",
        "packet_support": true,
        "size": 4580591172,
        "slab_fanout": 8,
        "slab_threshold": 131072,
      }
    }
    style={
      Object {
        "left": 1,
        "top": 1,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bounds": true,
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bounds": true,
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
